File: src/main/java/com/uitnetwork/alexa/service/IntroductionService.java
package com.uitnetwork.alexa.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.amazon.speech.json.SpeechletRequestEnvelope;
import com.amazon.speech.speechlet.LaunchRequest;
import com.amazon.speech.speechlet.SpeechletResponse;

@Service
public class IntroductionService {
    public static final String INTRODUCTION_MESSAGE = "Welcome to the Alexa Skills Kit, you can start by saying your name";
    public static final String INTRODUCTION_REPROMPT = "Tell me your name!";

    @Autowired
    private SpeechletResponseService speechletResponseService;

    public SpeechletResponse createIntrodutionSpeechletResponse(SpeechletRequestEnvelope<LaunchRequest> launchRequestEnvelope) {
        return speechletResponseService.newAskResponse(INTRODUCTION_MESSAGE, INTRODUCTION_REPROMPT);
    }
}

File: src/main/kotlin/com/uitnetwork/alexa/service/IntroductionService.kt
package com.uitnetwork.alexa.service

import com.amazon.speech.json.SpeechletRequestEnvelope
import com.amazon.speech.speechlet.LaunchRequest
import com.amazon.speech.speechlet.SpeechletResponse
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class IntroductionService {
    companion object {
        val INTRODUCTION_MESSAGE = "Welcome to the Alexa Skills Kit, you can start by saying your name"
        val INTRODUCTION_REPROMPT = "Tell me your name!"
    }

    @Autowired
    private lateinit var speechletResponseService: SpeechletResponseService

    fun createIntrodutionSpeechletResponse(launchRequestEnvelope: SpeechletRequestEnvelope<LaunchRequest>): SpeechletResponse {
        return speechletResponseService.newAskResponse(INTRODUCTION_MESSAGE, INTRODUCTION_REPROMPT)
    }
}
